src/main.c

```diff
@@ -150,12 +150,76 @@ bool is_appimage(char *path, gboolean verbose)
     return appimage_get_type(path, verbose) != -1;
 }
 
-void *thread_appimage_register_in_system(void *arguments)
-{
-    struct arg_struct *args = arguments;
-    if(is_appimage(args->path, args->verbose) && !appimage_is_registered_in_system(args->path)) {
-        appimage_register_in_system(args->path, args->verbose);
-        update_desktop_set_dirty();
+
+GKeyFile* load_desktop_entry(const char* desktop_file_path) {
+    GKeyFile* key_file_structure = g_key_file_new();
+    gboolean success = g_key_file_load_from_file(key_file_structure, desktop_file_path,
+                                                 G_KEY_FILE_KEEP_COMMENTS | G_KEY_FILE_KEEP_TRANSLATIONS, NULL);
+
+    if (!success) {
+        // Don't remove the dbrackets or the macro will segfault
+        THREADSAFE_G_PRINT("Failed to load the deployed desktop entry, '%s'\n", "a");
+    }
+
+
+    return key_file_structure;
+}
+
+void setup_firejail_on_desktop_entry(GKeyFile* key_file_structure) {
+    char* firejail_exec;
```

Comment:
There is no reason to declare the variable here and not next to its initialization.

Reply:
fixed